ebmdatalab/maps.py

```diff
@@ -1,131 +1,124 @@
-        map_file = data_dir / "ccgs{}_{}.json".format(cartogram_suffix, map_year)
-    else:
-        map_file = sorted(
-            glob.glob(str(data_dir / "ccgs{}_2*.json".format(cartogram_suffix)))
-        )[-1]
-    ccgs = gpd.read_file(map_file)
-    # Normalise names to match `ccg_fo_map` format (above)
-    ccgs["name"] = ccgs["name"].str.upper()
-    ccgs = ccgs.set_index("name")
-    # Remove ones without geometry - these are (usually) federations
-    # rather than individual CCGs
-    ccgs = ccgs[~ccgs["geometry"].isnull()]
-
-    # Check we can map all the CCGs named in the input data
-    diff = np.setdiff1d(df.index, ccgs.index)
-    if len(diff) > 0:
-        raise BaseException("Data contains CCG names we can't map: {}".format(diff))
-
-    # Join map with data
-    gdf = ccgs.join(df, rsuffix="_orig")
-
-    # Split into london and rest of England
-    gdf_london = gdf[gdf["is_london"] == True]
-    gdf_roe = gdf[gdf["is_london"] == False]
-
-    # set common value limits for colour scale
-    default_plot_options = {
-        'vmin': df[column].min(),
-        'vmax': df[column].max(),
-        'edgecolor': "black",
-        'linewidth': 0.1,
-        'cmap': "OrRd",
-    }
-
-    if plot_options is None:
-        plot_options = {}
-
-    for k, v in default_plot_options.items():
-        if k not in plot_options:
-            plot_options[k] = v
-
-    def plot(gdf, ax, title="", legend=True):
-        gdf.plot(
-            ax=ax,
-            column=column,
-            legend=legend,
-            **plot_options
-        )
-        ax.set_aspect(1.63)
-        if title:
-            ax.set_title(title, size=14)
-        ax.axis("off")
-
-    fig = plt.gcf()
-    if not subplot_spec:
+
+    def __init__(self, description=None, author="EBM DataLab", *args, **kwargs):
+        """Map Loader is initiated with description and author for clarity
+
+            Args:
+                description (str): Description at a broad level of the dataset
+                author (str): Author - set as EBM Datalab if left blank
+
+        """
+
+        self.description = description
+        self.author = author
+
+        self.date = datetime.datetime.now()
+        """Datetime (obj): Automatically records datetime as time class is initiated"""
+
+        self.plots = {}
+        """dict: initally blank to allow plots to be saved"""
+
+    def load_data(self, data, mapping_data_path, index_file_path=None):
+        """
+        Loads data from a pandas dataframe and combines with a mapping json file
+
+        :param data: pd.dataframe file
+        :param mapping_data_path (str): the path of the json file
+        :param index_file_path (str): path of the index file. Index file contains additionally information
+                about the geographical areas such as London False/True
+
+        :return: self.final_df
+        """
+
+        self.map_data = gpd.read_file(mapping_data_path)
+        if index_file_path == None:
```

Comment:
This is pushing the boundaries of nitpickery, but there's convention that comparisons with `None` should use `is` rather that `==` so this would ideally be:
```python
if index_file_path is None:
```

If you care about the details [this article](https://treyhunner.com/2019/03/unique-and-sentinel-values-in-python/) is a reasonable explanation. But probably not worth worrying about for now.

Reply:
cool thanks